pkg/objectupdate/rte/rte_test.go

```diff
@@ -179,6 +179,16 @@ func TestUpdateDaemonSetArgs(t *testing.T) {
 }
 
 func expectCommandLine(t *testing.T, ds, origDs *appsv1.DaemonSet, testName string, expectedArgs []string) {
+	uncontrolledArgs := []string{
+		//	"--notify-file=/run/rte/notify",
+		"--podresources-socket=unix:///host-var/lib/kubelet/pod-resources/kubelet.sock",
+		"--sysfs=/host-sys",
+		"--topology-manager-policy=restricted",
+		"--topology-manager-scope=pod",
+	}
+	if len(expectedArgs) != len(ds.Spec.Template.Spec.Containers[0].Args)-len(uncontrolledArgs) {
```

Comment:
this is  good idea, but then let's go full throttle.
1. we can rename 'uncontrollerdArgs' to `commonArgs`, or any better name which conveys the fact this is a subset of arguments we want to be there
2. turn the two slices into sets, adding `commonArgs` to any set to ensure the two sets contains them. The sets semantics ensure we don't have duplicates
3. turn the two sets into **sorted** lists
4. compare the two resulting sorted lists. If they aren't equal, fail the test

PS: please remove the commented out code :)

Reply:
thanks for your detailed comment. This is indeed a good idea if duplicates are not important. If sets are used and there happen to be a duplicate the conversation should fail not succeed and continue. one example is if an arg is duplicated then something is fishy and could turn out that in some place an arg was set while the intention is to set some other arg. in this case we want to fail.
However I do like adopting appending the slices. I've updated the code let me know what do you think. As for sorting slices I don't think that is helping a lot we anyway do a deepEqual and want enough debugging logs (like what arg was missing) in case this failed.

Reply:
sounds like a good easy check to exclude duplicates, I'll add it. thanks for pointing that out.